src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,175 @@
+        int taskIndex;
+        try {
+            taskIndex = Integer.parseInt(taskNumbers[0]);
+        } catch (Exception e){
+            throw new TaskNumberException();
+        }
+        if(taskIndex <= 0 || taskIndex > totalTasks){
+            throw new TaskNumberException();
+        }
+
+        // Minus one from task index since the array tasks is 0-based
+        taskIndex -= 1;
+        return taskIndex;
+    }
+    public static void changeTaskStatus(ArrayList<Task> tasks, String[] taskNumbers, boolean isDone) throws DukeException{
+        // From parser, taskNumbers.length is always equal to 1
+        int taskIndex = getTaskIndex(taskNumbers, tasks.size());
+
+        if(isDone){
+            if(tasks.get(taskIndex).isDone()){
+                System.out.println("This task is already marked done:");
+                System.out.println(tasks.get(taskIndex).toString());
+            } else{
+                tasks.get(taskIndex).setDone(true);
+                System.out.println("Nice! I've marked this task as done:");
+                System.out.println(tasks.get(taskIndex).toString());
+            }
+        }else{
+            if(!tasks.get(taskIndex).isDone()){
+                System.out.println("This task is already marked as not done:");
+                System.out.println(tasks.get(taskIndex).toString());
+            } else{
+                tasks.get(taskIndex).setDone(false);
+                System.out.println("OK, I've marked this task as not done yet:");
+                System.out.println(tasks.get(taskIndex).toString());
+            }
+        }
+    }
+
+    public static void deleteTask(ArrayList<Task> tasks, String[] taskNumbers) throws DukeException{
+        int taskIndex = getTaskIndex(taskNumbers, tasks.size());
+
+        Task taskToRemove = tasks.get(taskIndex);
+        tasks.remove(taskIndex);
+        System.out.println("Noted. I've removed this task:");
+        System.out.println("  " + taskToRemove);
+        System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+    }
+
+    public static void main(String[] args) {
+        ArrayList<Task> tasks;
+        final File taskDataDirectory = new File("data");
+        final File taskDataFile = new File("data/task-data.txt");
+        if(!taskDataDirectory.exists()){
+            try{
+                taskDataDirectory.mkdir();
+            }catch (SecurityException e){
+                System.out.println("An error occurred while creating the data directory: " + e.getMessage());
+                return;
+            }
+        }
+        try{
+            tasks = TaskData.retrieveTaskData(taskDataFile);
+        }catch (Exception e){
+            System.out.println("An error occurred while retrieving task data: " + e.getMessage());
+            return;
+        }
+        Scanner in = new Scanner(System.in);
+        printGreeting();
+        boolean isProgramRunning = true;
+        while(isProgramRunning)
+        {
```

Comment:
Try to keep your brace formatting consistent, in this case, the opening brace should be on the same line as `while(isProgramRunning)`